File: app/models/employee.py
from sqlalchemy import Boolean
from sqlalchemy import Column
from sqlalchemy import DateTime
from sqlalchemy import ForeignKey
from sqlalchemy import Integer
from sqlalchemy import String
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database import Base


class Employee(Base):

    __tablename__ = "employees"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    employee_code = Column(
        String(20),
        unique=True,
        nullable=False
    )

    name = Column(
        String(100),
        nullable=False
    )

    phone = Column(
        String(20),
        unique=True,
        nullable=False
    )

    address = Column(
        String(255),
        nullable=True
    )

    role = Column(
        String(50),
        nullable=False
    )

    route_id = Column(
        Integer,
        ForeignKey("routes.id"),
        nullable=True
    )

    is_active = Column(
        Boolean,
        default=True,
        nullable=False
    )

    user_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=True
    )

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now()
    )

    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now()
    )

    route = relationship(
        "Route",
        back_populates="employees"
    )

    user = relationship(
        "User"
    )

    @property
    def username(self):
        return self.user.username if self.user else None

File: frontend/src/types/employee.ts
export interface EmployeeBase {
  name: string;
  phone: string;
  address?: string | null;
  role: string;
  route_id?: number | null;
}

export interface EmployeeCreate extends EmployeeBase {
  username?: string | null;
  password?: string | null;
  confirm_password?: string | null;
}

export interface EmployeeUpdate {
  name?: string | null;
  phone?: string | null;
  address?: string | null;
  role?: string | null;
  route_id?: number | null;
}

export interface EmployeeCredentialsUpdate {
  username?: string | null;
  password?: string | null;
  confirm_password?: string | null;
}

export interface EmployeeResponse extends EmployeeBase {
  id: number;
  employee_code: string;
  is_active: boolean;
  username?: string | null;
  created_at: string;
  updated_at: string;
}

export interface EmployeeSummaryResponse {
  id: number;
  employee_code: string;
  name: string;
  phone: string;
  role: string;
}
